### PR TITLE
Add SendGrid to Readme and Task Library overview

### PR DESCRIPTION
**Pull request by taljaards**
## Summary
Add SendGrid to Readme and Task Library overview

## Changes
Adds SendGrid logo and links, which I missed in my initial PR.

## Importance
User experience and satisfaction 🤷‍♂️

## Checklist

This PR:

- [x] adds new tests (**not** appropriate)
- [x] adds a change file in the `changes/` directory
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs  (**not** appropriate)

### CODE REVIEW

File: changes/pr4670.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+fix:
+  - "Add SendGrid to Readme and Task Library overview - [#4670](https://github.com/PrefectHQ/prefect/pull/4670)"
```
Comment (madkinsz) [inline changes/pr4670.yaml:2]: We generally don't do changelog entries for docs updates ie this doesn't fix an actual bug users need to worry about

**Review by madkinsz — COMMENTED** [review]
Other than the changelog entry, this lgtm! Adding images to that table on the README is certainly a pain, looks like you did a nice job though!

File: changes/pr4670.yaml (hunk 1 above)
Comment (taljaards) [reply changes/pr4670.yaml:2]: Sounds good, I'll remove that file.

**Comment by taljaards** [comment]
@madkinsz I used the [Markdown Table Prettifier](https://marketplace.visualstudio.com/items?itemName=darkriszty.markdown-table-prettify) VS Code extension to format the tables. Seems to work quite well.

**Review by madkinsz — APPROVED** [review]

**Pull request merged**